### PR TITLE
Split `RuntimeType` and `CargoDependency` for server code generation

### PR DESCRIPTION
**Pull request by david-perez**
For clear separation of concerns.

## Checklist
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated `aws/SDK_CHANGELOG.md` if applicable

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by david-perez** [comment]
This addresses https://github.com/awslabs/smithy-rs/pull/850#discussion_r750476807

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/CargoDependency.kt [hunk 1]
```diff
@@ -206,7 +204,6 @@ data class CargoDependency(
         val Ring: CargoDependency = CargoDependency("ring", CratesIo("0.16"))
         val TempFile: CargoDependency = CargoDependency("temp-file", CratesIo("0.1.6"), scope = DependencyScope.Dev)
         val TokioStream: CargoDependency = CargoDependency("tokio-stream", CratesIo("0.1.7"))
-        val Tower: CargoDependency = CargoDependency("tower", CratesIo("0.4"), optional = true)
```
Comment (crisidev) [inline codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/CargoDependency.kt:209]: Was tower unused?

**Review by crisidev — APPROVED** [review]
LGTM

File: codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/CargoDependency.kt (hunk 1 above)
Comment (david-perez) [reply codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/rustlang/CargoDependency.kt]: 😅

**Review by jdisanti — APPROVED** [review]

**Pull request merged**